dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/datastore/FileDataStoreHandler.kt

```diff
@@ -0,0 +1,289 @@
+        val versionCodeBlock = typesToBlocks[TLVBlockType.VERSION_CODE]
+        val dataBlock = typesToBlocks[TLVBlockType.DATA]
+        return if (lastUpdateBlock == null || versionCodeBlock == null || dataBlock == null) {
+            null // this should never happen as we know by this stage that these cannot be null
+        } else {
+            DataStoreContents(
+                lastUpdateDate = lastUpdateBlock.data.toLong(),
+                versionCode = versionCodeBlock.data.toInt(),
+                data = deserializer.deserialize(String(dataBlock.data))
+            )
+        }
+    }
+
+    private fun isDataStoreTooOld(lastUpdateDate: Long): Boolean {
+        val currentTime = System.currentTimeMillis()
+        return currentTime - lastUpdateDate > DATASTORE_EXPIRE_TIME
+    }
+
+    private fun logSameBlockAppearsTwiceError(type: TLVBlockType) {
+        internalLogger.log(
+            target = InternalLogger.Target.MAINTAINER,
+            level = InternalLogger.Level.ERROR,
+            messageBuilder = { SAME_BLOCK_APPEARS_TWICE_ERROR.format(Locale.US, type) }
+        )
+    }
+
+    private fun logInvalidVersionError() {
+        internalLogger.log(
+            level = InternalLogger.Level.ERROR,
+            target = InternalLogger.Target.MAINTAINER,
+            messageBuilder = { INVALID_VERSION_ERROR }
+        )
+    }
+
+    private fun logInvalidNumberOfBlocksError(numberOfBlocks: Int) {
+        internalLogger.log(
+            level = InternalLogger.Level.ERROR,
+            target = InternalLogger.Target.MAINTAINER,
+            messageBuilder = { INVALID_NUMBER_OF_BLOCKS_ERROR.format(Locale.US, numberOfBlocks) }
+        )
+    }
+
+    private fun logFailedToSerializeDataError() {
+        internalLogger.log(
+            target = InternalLogger.Target.MAINTAINER,
+            level = InternalLogger.Level.ERROR,
+            messageBuilder = { FAILED_TO_SERIALIZE_DATA_ERROR }
+        )
+    }
+
+    private fun createDataStoreDirectoryIfNecessary(featureName: String): File {
+        val dataStoreDirectory = dataStoreFileHelper.getDataStoreDirectory(
+            sdkInstanceId = sdkInstanceId,
+            featureName = featureName,
+            folderName = DATASTORE_FOLDER_NAME.format(Locale.US, CURRENT_DATASTORE_VERSION),
+            storageDir = storageDir
+        )
+
+        if (!dataStoreDirectory.existsSafe(internalLogger)) {
+            dataStoreDirectory.mkdirsSafe(internalLogger)
+        }
+
+        return dataStoreDirectory
+    }
+
+    private fun createDataStoreFileIfNecessary(
+        dataStoreDirectory: File,
+        dataStoreFileName: String
+    ): File {
+        val datastoreFile = dataStoreFileHelper.getDataStoreFile(
+            dataStoreDirectory = dataStoreDirectory,
+            dataStoreFileName = dataStoreFileName
+        )
+
+        if (!datastoreFile.existsSafe(internalLogger)) {
+            datastoreFile.createNewFileSafe(internalLogger)
+        }
```

Comment:
this is not needed, I think. Once you have a `File` object, underlying file will be created automatically once write operation is executed.

This leads to the removal of this method, because then the only thing it does is calling `getDataStoreFile`